libs/js-sdk/src/create-browser-client.ts

```diff
@@ -41,29 +45,18 @@ export function createBrowserClient({
 
     environmentApiKey: string
 }): BrowserClient {
-    const initialPromise = new PromiseCompletionSource<boolean>()
+    const tracer = getTracer()
```

Comment:
This is probably the riskiest change in the PR.

I rewrite the retry logic and removed a bunch of duplication in this function based on the traces not quite looking right.